structuralcodes/codes/ec2_2023/_section5_materials.py

```diff
@@ -0,0 +1,1118 @@
+    if fy <= 0:
+        raise ValueError(f'fyk={fy} must be larger than 0')
+    if k < 0:
+        raise ValueError(f'k={k} must be larger than 1')
+
+    eps_y = fy / _Es
+
+    # If in elastic area
+    if eps <= eps_y:
+        return eps * _Es
+
+    # If in plastic area
+    m = fy * (k - 1) / (eps_u - eps_y)
+    return fy + m * (eps - eps_y)
+
+
+def p_steel_stress_strain_params(
+    prestress_class: str, element: str
+) -> t.Tuple[float, float]:
+    """Computes the stress-strain-diagram parameters
+    fp01k and fpk
+
+    FprEN 1992-1-1, 5.3.3
+
+    Args:
+        prestress_class (str): possible values: Y1560, 1670,
+            Y1770, Y1860, Y1770, Y1860, Y1960, Y2060,
+            Y1030, Y1050, Y1100 and Y1230
+        element (str):
+            'W' for Wires
+            'S' for Strands
+            'B' for Bars
+
+    Returns:
+        Tuple(float, float): with the value of fp01k and fpk in MPa
+
+    Raises:
+        ValueError: if combination of prestress_class and element
+            is not a possible value from the range
+    """
+    prestress_class = prestress_class.upper().strip()
+    element = element.upper().strip()
+    data = {
+        'W': {
+            'Y1570': (1380, 1570),
+            'Y1670': (1470, 1670),
+            'Y1770': (1550, 1770),
+            'Y1860': (1650, 1860),
+        },
+        'S': {
+            'Y1770': (1560, 1770),
+            'Y1860': (1640, 1860),
+            'Y1960': (1740, 1960),
+            'Y2060': (1830, 2060),
+        },
+        'B': {
+            'Y1030': (835, 1030),
+            'Y1050': (950, 1050),
+            'Y1100': (900, 1100),
+            'Y1230': (1080, 1230),
+        },
+    }
+
+    if element not in data:
+        raise ValueError(f'element={element} has not a valid value')
+
+    material = data[element]
+
+    if prestress_class not in material:
+        raise ValueError(
+            f'prestress_class={prestress_class} has not a valid value'
+        )
+
+    return material[prestress_class]
+
+
+def fpd(fp01k: float, gamma_S: float) -> float:
+    """Computes the design value for the prestressing steel stress
+
+    FprEN 1992-1-1, 5.3.3
+
+    Args:
+        fp01k (float): the 0.1% proof stress in MPa
+        gamma_S (float): the safety coefficient
+
+    Returns:
+        float: the design value for the design prestressing
+            steel stress in MPa
+
+    Raises:
+        ValueError: if fp01k is less than 0
+        ValueError: if gamma_S is less or equal to 0
+    """
+    if fp01k < 0:
+        raise ValueError(f'fp01k={fp01k} must be larger or equal to 0')
+    if gamma_S <= 0:
+        raise ValueError(f'gamma_S={gamma_S} must be larger than 0')
+
+    return fp01k / gamma_S
+
+
+def Ep() -> float:
+    """Returns the modulus of elasticity for prestressing steel
+    in MPa
+
+    Returns:
+        float: modulus of elasticity in MPa
+    """
+    return 200000
+
+
+def sigma_p(
+    eps: float,
```

Comment:
Indeed, you are correct, we could use the same function for both. Nevertheless, I created two functions for helping the user with the data. Let's take a look.

The signature for computing the stress in the reinforcing steel is as follows:
```
def sigma_s(eps: float, fy: float, k: float, eps_u: float, _Es: float = 200000) -> float:
```
![image](https://github.com/fib-international/structuralcodes/assets/80036012/560ab7c1-442a-47a8-9c77-209f720ca34a)

And for the prestressing steel:
```
def sigma_p(eps: float, fpy: float, fpu: float, eps_u: float = 0.035, Ep: float = 190000) -> float: 
```
![image](https://github.com/fib-international/structuralcodes/assets/80036012/13e8c1dc-be90-4418-aed7-c0b7ed87b21a)

The curves are the same, but we use a parameter `k` in the reinforcing steel curve and `fpu`in the prestressing steel. Just because I wanted to help the final user to use these functions, I created both. 

Paremeters for both curves can be also be obtained with the functions `p_steel_stress_params` and `r_steel_stress_strain_params`.

We can discuss it if you want